tests/test_fixtures_edges.py

```
@@ -0,0 +1,111 @@
+
+def test_bounded_jpeg_compresses_an_oversized_image_to_fit() -> None:
+    # A huge, high-entropy image never fits at any JPEG quality, forcing
+    # the quality-step loop AND the resize fallback before it returns.
+    import random
+
+    image = Image.new("RGB", (6000, 6000))
+    random.seed(7)
+    pixels = image.load()
+    for y in range(0, 6000, 2):
+        for x in range(0, 6000, 2):
+            pixels[x, y] = (random.randrange(256), random.randrange(256), random.randrange(256))
```

Ruff (0.16.2)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)